server/container_events.go

```diff
@@ -1,17 +1,76 @@
 package server
 
 import (
+	"sync"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 	types "k8s.io/cri-api/pkg/apis/runtime/v1"
 )
 
-// GetContainerEvents sends the stream of container events to the client - kubelet
-func (s *Server) GetContainerEvents(req *types.GetEventsRequest, ces types.RuntimeService_GetContainerEventsServer) error {
-	if s.Config().EnablePodEvents {
-		for containerEvent := range s.ContainerEventsChan {
-			if err := ces.Send(&containerEvent); err != nil {
-				return err
-			}
+type containerEventConn struct {
+	Wg  sync.WaitGroup
+	Err error
```

Comment:
Thanks for the PR @Freggy !!
Changes LGTM expect that do we need to export these fields if not used in any other package?

Reply:
No problem, I'm hoping to contribute more in the future :) 
There is no need for those fields being public, so I made them private.